contracts/cw-proposal-single/src/contract.rs

```diff
@@ -846,15 +850,54 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
                 }
                 None => Err(ContractError::NoSuchProposal { id: proposal_id }),
             })?;
-            Ok(Response::new().add_attribute("proposal execution failed", proposal_id.to_string()))
+
+            Ok(Response::new().add_attribute("proposal_execution_failed", proposal_id.to_string()))
         }
         TaggedReplyId::FailedProposalHook(idx) => {
-            let addr = PROPOSAL_HOOKS.remove_hook_by_index(deps.storage, idx)?;
-            Ok(Response::new().add_attribute("removed proposal hook", format!("{addr}:{idx}")))
+            // When firing off proposal hooks indexes `[0,
+            // hook_count)` are used to represent hooks that have been
+            // fired via the regular PROPOSAL_HOOKS method. Index
+            // `hook_count` is used to represent the pre-propose
+            // module.
+            let count = PROPOSAL_HOOKS.hook_count(deps.storage)?;
```

Comment:
can you explain why it is the case that `hook_count` will correspond to the pre-propose module?

Reply:
we talked about separating out prepropose from the proposal hooks